shards/shards.go

```diff
@@ -628,6 +629,8 @@ func streamSearch(ctx context.Context, proc *process, q query.Q, opts *zoekt.Sea
 
 	defer cancel()
 
+	// We set the number of workers to GOMAXPROCS, or the number of shards,
+	// whichever is smaller.
```

Comment:
nit: this is the only comment which doesn't explain why, but is just restating the code.